scripts/agent_app.py

```diff
@@ -118,39 +145,35 @@ def agent_app() -> None:
         "Skip market verification", value=False, key="skip_market_verification"
     )
 
-    for idx, (column, AgentClass) in enumerate(
-        zip(st.columns(len(agent_classes)), agent_classes)
-    ):
-        with column:
-            agent = AgentClass(
-                place_bet=False,
-                enable_langfuse=APIKeys().default_enable_langfuse
-                and not custom_question_input  # Don't store custom inputs, as we won't have true data for them.
-                and not skip_market_verification,  # Don't store unverified markets, as they wouldn't be predicted anyway.
-            )
-            start_time = time.time()
-
-            # Show the agent's title.
-            st.write(
-                f"## {agent.__class__.__name__.replace('Deployable', '').replace('Agent', '')}"
-            )
-
-            with st.spinner("Agent is processing the market..."):
-                processed_market = agent.process_market(
-                    market_source, market, verify_market=not skip_market_verification
-                )
+    # Select an agent from the list of available agents.
```

Comment:
I moved this down so we can first select a question and then an agent. Otherwise, the agent starts predicting right away some random question that we didn't even want.